win32ss/user/ntuser/class.c

```diff
@@ -1275,39 +1300,49 @@ IntGetAtomFromStringOrAtom(
     if (ClassName->Length != 0)
     {
         WCHAR szBuf[65];
-        PWSTR AtomName;
-        NTSTATUS Status;
+        PWSTR AtomName = szBuf;
+        NTSTATUS Status = STATUS_INVALID_PARAMETER;
 
         *Atom = 0;
 
         /* NOTE: Caller has to protect the call with SEH! */
-
-        if (ClassName->Length != 0)
+        if (ClassName->Length / sizeof(WCHAR) >= ARRAYSIZE(szBuf))
         {
-            /* FIXME: Don't limit to 64 characters! use SEH when allocating memory! */
-            if (ClassName->Length / sizeof(WCHAR) >= sizeof(szBuf) / sizeof(szBuf[0]))
+            AtomName = ExAllocatePoolWithTag(PagedPool,
+                                                ClassName->Length + sizeof(UNICODE_NULL),
+                                                TAG_STRING);
+            if (AtomName == NULL)
             {
-                EngSetLastError(ERROR_INVALID_PARAMETER);
-                return (RTL_ATOM)0;
+                EngSetLastError(ERROR_OUTOFMEMORY);
+                return FALSE;
             }
+        }
 
-            /* We need to make a local copy of the class name! The caller could
-               modify the buffer and we could overflow in RtlLookupAtomInAtomTable.
-               We're protected by SEH, but the ranges that might be accessed were
-               not probed... */
-            RtlCopyMemory(szBuf,
+        /* We need to make a local copy of the class name! The caller could
+            modify the buffer and we could overflow in RtlLookupAtomInAtomTable.
+            We're protected by SEH, but the ranges that might be accessed were
+            not probed... */
```

Comment:
Again, why would this throw? This shouldn't be user mode memory. And if it is, why is it not probed?

Reply:
This comment was here, should I delete it + the seh?